Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -524,4 +529,73 @@ public int getLATValue(Object adInfoObj) {
         }
         return LATVal_;
     }
+
+    /**
+     * <p>
+     * Method to prefetch the GAID and LAT values
+     * </p>
+     *
+     * @param callback {@link GAdsParamsFetchEvents} instance to notify process completion
+     * @return {@link Boolean} with true if GAID fetch process started.
+     */
+    public boolean prefetchGAdsParams(GAdsParamsFetchEvents callback) {
+        boolean isPrefetchStarted = false;
+        if (TextUtils.isEmpty(GAIDString_)) {
+            isPrefetchStarted = true;
+            new GAdsPrefetchTask(callback).execute();
+        }
+        return isPrefetchStarted;
+    }
+
+    /**
+     * <p>
+     * Async task to fetch GAID and LAT value
+     * This task fetch the GAID and LAT in background. The Background task times out
+     * After GAID_FETCH_TIME_OUT
+     * </p>
+     */
+    private class GAdsPrefetchTask extends AsyncTask<Void, Void, Void> {
```

Comment:
Do we need to get rid off the Async task. I found it easier to switch to
main thread than "runOnUiThread" which always requires an "activity"
context. When we start this task there no activity started in most cases
On Jul 11, 2016 6:50 PM, "Alex Austin" <notifications@github.com> wrote:

> In Branch-SDK/src/io/branch/referral/SystemObserver.java
> <https://github.com/BranchMetrics/android-branch-deep-linking/pull/267#discussion_r70366774>
> :
>
> > +        boolean isPrefetchStarted = false;
> > +        if (TextUtils.isEmpty(GAIDString_)) {
> > +            isPrefetchStarted = true;
> > +            new GAdsPrefetchTask(callback).execute();
> > +        }
> > +        return isPrefetchStarted;
> > +    }
> > +
> > +    /**
> > +     * <p>
> > +     * Async task to fetch GAID and LAT value
> > +     * This task fetch the GAID and LAT in background. The Background task times out
> > +     * After GAID_FETCH_TIME_OUT
> > +     * </p>
> > +     */
> > +    private class GAdsPrefetchTask extends AsyncTask<Void, Void, Void> {
>
> Nope. Never mind. Can't you do what I describe above then call the
> callback on the main thread?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BranchMetrics/android-branch-deep-linking/pull/267/files/5a1edc5d32f01bf9cb320a7fadff292350a47b7e#r70366774>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHTA9HzqDQnZP4gMskzoK6b9uqEiXKUKks5qUvMAgaJpZM4JJ7UG>
> .
>